src/target/riscv/batch.h

```diff
@@ -190,14 +190,32 @@ int riscv_batch_run_from(struct riscv_batch *batch, size_t start_idx,
 size_t riscv_batch_finished_scans(const struct riscv_batch *batch);
 
 /* Adds a DM register write to this batch. */
-void riscv_batch_add_dm_write(struct riscv_batch *batch, uint64_t address, uint32_t data,
+void riscv_batch_add_dmi_write(struct riscv_batch *batch, uint64_t address, uint32_t data,
 	bool read_back, enum riscv_scan_delay_class delay_class);
 
+static inline void
+riscv_batch_add_dm_write(struct riscv_batch *batch, uint64_t address, uint32_t data,
+	bool read_back, enum riscv_scan_delay_class delay_type)
+{
+	return riscv_batch_add_dmi_write(batch,
+			riscv_get_dmi_address(batch->target, address), data,
+			read_back, delay_type);
+}
+
```

Comment:
[Just a note - no change requested]

I tend to not use static inline functions in header files. For the sake of unified code style, I prefer the usual layout where possible: Prototype in header file and implementation in the corresponding .c file.

I'd also leave the optimizations for the compiler (avoid micro-optimization) unless there is a significant gain.

Reply:
My intention here is not to micro-optimize. The main point of declaring `_dm_` functions together their definitions is to highlight that the only difference from `_dmi_` counterparts is the address conversion.

Reply:
Hmm, ok, so the reason for `static inline` here is to make it very clear that the `_dm_` function is almost the same thing (a simple wrapper) over the `_dmi_` function. 

I'd still prefer to not use static inline, but that's a detail and matter of personal taste.

So let's leave it as is.